### ISSUE TITLE
[SC][TEST]: Manage focus on review and submit page after clicking edit

### ISSUE THREAD

**Issue by davidakennedy** [department-of-veterans-affairs/va.gov-team#92801]
## What happened?

**Describe what went wrong**:

When you click on many of the edit buttons on the review and submit page, focus is lost.

**Add the steps to reproduce the bug**:

1. Go to `https://staging.va.gov/decision-reviews/test-sc/` and start filling out the form.
2. Navigate to `https://staging.va.gov/decision-reviews/test-sc/review-and-submit`
3. Expand the accordions.
4. Activate one of the edit buttons.
5. Note that focus is lost.

**Add a screen recording or photo**:

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL1RRQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3e4db69c9b552b4d445e1224b6446ed23d507cd2/Screen%20Recording%202024-09-12%20at%205.55.24%E2%80%AFPM.mov)

## How severe is the problem?

- [x] **Critical**: User is block or significantly hindered from submitting the form, data is lost, or claim and evidence test coverage is missing. Fix ASAP.
- [ ] **Moderate**: User is not blocked, but the happy path is interrupted. Fix after critical bugs.
- [ ] **Minor**: Cosmetic bugs and minor language inconsistencies. Defer to a later sprint.

## What is the desired behavior?

1. _When a user activates the edit button on of a question, the **visual** result is focus is shifted._

We should shift focus to the `input` field of the associated question. If there isn't an input, to a heading or first line of text.

## Definition of Done

- [ ] Acceptance criteria reviewed and approved by product or design
- [ ] Unit test (or other test coverage) added if possible
- [ ] Logging and monitoring in place

**Comment by Mottie** [comment]
This appears to be a duplicate of #74925 where our focus functions aren't working as expected on nested web components. I'll go ahead and continue work on this problem.

**Comment by eileen-coforma** [comment]
Still in progress

**Comment by davidakennedy** [comment]
@Mottie Tested all the working review sections on the review and submit page in the prototype form. Everything works as expected, and I do like the consistency of moving focus to the headings.
